pkg/controller/azure/storage/container/claim.go

```diff
@@ -53,13 +53,11 @@ func (c *ClaimController) SetupWithManager(mgr ctrl.Manager) error {
 
 	name := strings.ToLower(fmt.Sprintf("%s.%s", storagev1alpha1.BucketKind, controllerName))
 
-	p := v1alpha1.ContainerKindAPIVersion
-
 	return ctrl.NewControllerManagedBy(mgr).
 		Named(name).
 		Watches(&source.Kind{Type: &v1alpha1.Container{}}, &resource.EnqueueRequestForClaim{}).
 		For(&storagev1alpha1.Bucket{}).
-		WithEventFilter(resource.NewPredicates(resource.ObjectHasProvisioner(mgr.GetClient(), p))).
+		WithEventFilter(resource.NewPredicates(resource.HasClassReferenceKind(resource.ClassKind(v1alpha1.ContainerClassGroupVersionKind)))).
```

Comment:
what's the latest thinking on the portability of Azure blob storage? Can we not use the portable Bucket abstraction here like we do for AWS and GCP?

What else do we have to do to cover Azure with the portable bucket abstraction?

Reply:
Oh oops, I goofed, we are already doing that in the line above:
```
For(&storagev1alpha1.Bucket{}).
```

LGTM

Reply:
This is definitely something we want to address. I think we likely will combine by just creating an `account` if one does not already exist when a `container` is created. The process currently looks like [this](https://github.com/crossplaneio/crossplane/tree/master/cluster/examples/storage/bucket/azure) where `Bucket` is an abstract type for both `account` and `container`. It is not currently a necessity to satisfy the features in epic #617 (specifically #649), but I would be happy to add it as an additional feature. Either way, I will make sure an issue is opened to address this concern.

Reply:
I agree that is not in scope for epic #617 in v0.3, but arriving at complete portability story for Azure's blob storage is definitely where we want to be long term. Maybe do open an issue on that, if there's not one already.  Maybe #117 is a good spot to capture this?